docs/orchestration.asciidoc

```diff
@@ -0,0 +1,104 @@
+* When an existing NodeSet is renamed, ECK performs the creation of a new NodeSet with the new name, and the removal of the old NodeSet, according to the NodeSet creation and removal patterns described above. Elasticsearch data is migrated away from the deprecated NodeSet before removal. The Elasticsearch resource <<{p}-update-strategy>> controls how many nodes can exist above or below the target node count during the upgrade.
+
+In any case, ECK ensures any StatefulSet operation is done in a way that is consistent with the Elasticsearch cluster orchestration best practices, by manipulating orchestration settings such as `discovery.seed_hosts`, `cluster.initial_master_nodes`, `discovery.zen.minimum_master_nodes`, `_cluster/voting_config_exclusions`.
+
+[id="{p}-orchestration-limitations"]
+==== Limitations
+
+Due to the way Kubernetes and StatefulSets behave, ECK orchestration has the following limitations:
+
+* Storage requirements (including volume size) of an existing NodeSet cannot be updated. link:https://github.com/kubernetes/enhancements/issues/661[StatefulSet volumes expansion is not available yet in Kubernetes]. In order to upgrade the storage size, it is however possible to create a new NodeSet, or rename the existing one. The rename will lead to the creation of a new StatefulSet with the specified storage size. The deprecated StatefulSet is removed once Elasticsearch data has been migrated away to nodes of the replacing one, automatically handled by ECK.
+* Cluster availability cannot be guaranteed during the rolling upgrade of a one-node cluster, or for a cluster having indices with no replicas. If an Elasticsearch node holds the only copy of a shard, this shard becomes unavailable while the node is upgraded. Clusters with more than one node and at least one replica per index are considered best practice.
+* Elasticsearch Pods may stay `Pending` during a rolling upgrade if the Kubernetes scheduler cannot re-schedule them back. This is especially important when using local PersistentVolumes. If the Kubernetes node bound to a local PersistentVolume does not have enough capacity to host an upgraded Pod which was temporarily removed, that Pod will stay Pending.
+* Rolling upgrades can only progress if the Elasticsearch cluster reports a `green` health. Upgrading a cluster reporting a `yellow` health may cause some shards to become unavailable, turning the cluster health to `red`. To prevent that situation, ECK waits for the cluster to become green in order to make progress. If strictly necessary, it is still possible for the user to manually delete Pods to upgrade. They will be automatically recreated in the newest revision. Note that in some cases where all nodes of a NodeSet are unavailable, ECK may still proceed to upgrading the corresponding Pods, even though the cluster does not report a `green health.`
```

Comment:
I would get rid of the `a` in front of the colors e.g. "green health" instead of "a green health". 

> If strictly necessary, it is still possible for the user to manually delete Pods to upgrade.

It is still possible for the user to force an upgrade by manually deleting Pods.

> They will be automatically recreated in the newest revision

Not sure if it should be  "They will be automatically recreated at the latest revision" Maybe @anyasabo can help as a native speaker?

Reply:
Not an expert, but "They will be automatically recreated at the latest revision" sounds better. Alternatively, "They will automatically be recreated using the latest revision" could work as well. If I were to write this paragraph, I would write it as:

> Rolling upgrades can only make progress if the Elasticsearch cluster health is `green`. It is risky to attempt upgrading a cluster in the `yellow` state as some shards could become completely unavailable and  degrade the cluster health to `red`. ECK takes the cautionary approach of waiting for `green` before progressing but advanced users may force an upgrade by manually deleting pods themselves. The deleted pods will be automatically recreated at the latest revision. 
> In the exceptional case that all the nodes of a node set are unavailable, the operator might choose to ignore the cluster health and continue with the upgrade anyway.

Reply:
>Note that in some cases where all nodes of a NodeSet are unavailable, ECK may still proceed to upgrading the corresponding Pods, even though the cluster does not report a `green health.`

This can also be changed to "Note that in the event that *all* nodes of a NodeSet..."

Reply:
I should have read everyone else's comments first: agreed with Charith that "at the latest revision" reads better to me. I would also recommend being more specific in the last sentence rather than "may still proceed" or "might choose"

Reply:
Thanks all!